chia/rpc/wallet_rpc_api.py

```diff
@@ -483,14 +483,31 @@ async def _check_key_used_for_rewards(
             return False, False
 
         config: Dict[str, Any] = load_config(new_root, "config.yaml")
-        farmer_target = config["farmer"].get("xch_target_address")
-        pool_target = config["pool"].get("xch_target_address")
-        address_to_check: List[bytes32] = [decode_puzzle_hash(farmer_target), decode_puzzle_hash(pool_target)]
+        farmer_target = config["farmer"].get("xch_target_address", "")
+        pool_target = config["pool"].get("xch_target_address", "")
+        address_to_check: List[bytes32] = []
+
+        try:
+            farmer_decoded = decode_puzzle_hash(farmer_target)
+            address_to_check.append(farmer_decoded)
+        except ValueError:
+            farmer_decoded = None
```

Comment:
Do we want all invalid addresses to be silently consumed and reported the same as valid addresses that we do not have the private keys for?

Reply:
Yes, at least in the one case (deleting a key) this function is currently used, it is expected. When deleting a key, all this is checking is that that one key isn't being used as the xch target address.

Currently, this function is only used when deleting a key.

(it is a reasonable question why/how something invalid might end up in xch_target_address - but that wasn't being addressed here - it should get filled in when a key is first created/imported)